### PR TITLE
Clean up CI jobs

### PR DESCRIPTION
**Pull request by gibizer**
* unify make test and gotest targets
* remove redundant github actions preferring their prow counterparts
* ensure image build can be fixed if fails 
* add a small check that prevents merging go.mod replace lines

### CODE REVIEW

**Comment by gibizer** [comment]
For the functional prow job to pass we need https://github.com/openshift/release/pull/35605 to land first

**Comment by gibizer** [comment]
/retest
the necessary CI change landed

File: .github/workflows/lints.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+jobs:
+  check-go-mod-replace-lines:
+    name: check for replace lines in go.mod files
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout project code
+        uses: actions/checkout@v3
+      - name: check for replace lines in go.mod files
+        run: |
+          ! egrep --invert-match -e '^replace.*/api => \./api|^replace.*//allow-merging$'  `find . -name 'go.mod'` | egrep -e 'go.mod:replace'
```
Comment (SeanMooney) [inline .github/workflows/lints.yaml:14]: jsut confirmign on irc but

egrep --invert-match -e '^replace.*/api => \./api|^replace.*//allow-merging$'

this is intennded to allow replaces that have a //allow-merging comment
and also to allow the API module to explicitly be referneced/replaced with the one form the current repo

**Review by SeanMooney — COMMENTED** [review]
the rest of this looks fine once you confirm my understanding is correct we can approve an merge this

File: .github/workflows/lints.yaml (hunk 1 above)
Comment (gibizer) [reply .github/workflows/lints.yaml:14]: yes

Comment (SeanMooney) [reply .github/workflows/lints.yaml:14]: cool ill approve so

**Pull request merged**